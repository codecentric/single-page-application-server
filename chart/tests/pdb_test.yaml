suite: test pdb
templates:
  - pdb.yaml
tests:
  - it: should not render pdb if disabled
    values:
      - ./values/minimal_values.yaml
    set:
      pod.disruptionBudget.enabled: false
    asserts:
      - hasDocuments:
          count: 0
  - it: should render pdb by default
    values:
      - ./values/minimal_values.yaml
    asserts:
      - matchSnapshot: { }
  - it: should fail if minAvailable >= replicas
    values:
      - ./values/minimal_values.yaml
    set:
      pod:
        replicas: 2
        disruptionBudget:
          minAvailable: 2
    asserts:
      - failedTemplate:
          errorMessage: "replicaCount must be greater than pod.disruptionBudget.minAvailable!"
  - it: should render with minAvailable
    values:
      - ./values/minimal_values.yaml
    set:
      pod:
        disruptionBudget:
          minAvailable: 1
          maxUnavailable: 0
    asserts:
      - matchSnapshot: { }
