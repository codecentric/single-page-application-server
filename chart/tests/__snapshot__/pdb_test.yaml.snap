should render pdb by default:
  1: |
    apiVersion: policy/v1
    kind: PodDisruptionBudget
    metadata:
      labels:
        app.kubernetes.io/component: frontend
        app.kubernetes.io/instance: default
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: RELEASE-NAME
        app.kubernetes.io/version: latest
        helm.sh/chart: single-page-application-server-0.0.0
      name: RELEASE-NAME
    spec:
      maxUnavailable: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: default
          app.kubernetes.io/name: RELEASE-NAME
should render with minAvailable:
  1: |
    apiVersion: policy/v1
    kind: PodDisruptionBudget
    metadata:
      labels:
        app.kubernetes.io/component: frontend
        app.kubernetes.io/instance: default
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: RELEASE-NAME
        app.kubernetes.io/version: latest
        helm.sh/chart: single-page-application-server-0.0.0
      name: RELEASE-NAME
    spec:
      minAvailable: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: default
          app.kubernetes.io/name: RELEASE-NAME
